src/gzchunked/mod.rs

```diff
@@ -99,16 +116,76 @@ mod tests {
     use super::*;
     use rand::{Rng, SeedableRng, rngs::StdRng};
 
+    #[test]
+    fn test_encode_and_decode_empty() {
+        let mut encoder = GzChunkedEncoder::new(GzChunkedCompression::default());
+        let mut decoder = GzChunkedDecoder::new();
+
+        let encoded_block = encoder.next_chunk().unwrap();
+        // should contain gzip header
+        assert_ne!(encoded_block.len(), 0);
+        decoder.write(encoded_block.as_slice()).unwrap();
+        decoder.write(encoded_block.as_slice()).unwrap();
+
+        assert!(decoder.try_next_data().is_none());
+    }
+
+    #[test]
+    fn test_encode_and_decode_all_in_one_block() {
+        let mut encoder = GzChunkedEncoder::new(GzChunkedCompression::default());
+        let mut decoder = GzChunkedDecoder::new();
+        let blocks = vec![
+            vec![1, 2, 3, 4],
+            vec![],
+            vec![1, 2, 3, 4, 5, 6, 7, 8],
+            vec![1],
+        ];
+
+        for block in &blocks {
+            encoder.write(block.as_slice()).unwrap();
+        }
+
+        let encoded_block = encoder.next_chunk().unwrap();
+        decoder.write(encoded_block.as_slice()).unwrap();
+
+        for block in blocks {
+            assert_eq!(block, decoder.try_next_data().unwrap());
+        }
```

Comment:
Here, it is not only readability issue but also debuggability. If the test fails, you don't really know at which iteration it failed (I guess in this case you could figure it out, but if someone changes the test data to be non-unique, then you have a problem again).

Writing test assertions like this is very tempting if you create variables, but if you follow the advice from the previous comment, then it is just not possible.